addons/mission/functions/fnc_earthquake.sqf

```diff
@@ -0,0 +1,88 @@
+#include "script_component.hpp"
+/*
+ * Author: Zozo (BI), Mike
+ * Rewrite of BIS_fnc_earthquake to run unscheduled and remove excess fatigue code.
+ * Uses magnitude between 1-5
+ *
+ * Call from init.sqf
+ *
+ * Arguments:
+ * 0: Magnitude <NUMBER> (default: 1)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [1] call TAC_Mission_fnc_earthquake;
+ */
+
+params [
+    ["_magnitude", 1],
+    ["_power", 0.3],
+    ["_duration", 20],
+    ["_frequency", 100],
+    ["_offset", 1],
+    ["_compensation", 0],
+    ["_eqSound", "Earthquake_01"]
+];
+
+if (GVAR(earthquakeInProgress)) exitWith {
+    if (is3DENPreview) then {
+        hint "[Earthquake]: Only one instance of Earthquake will happen locally.";
+    };
+};
+
+switch (_magnitude) do {
+    case 1: {
+        _power = 0.5;
+        _duration = 13;
+        _compensation = 4;
+        _frequency = 200;
+    };
+    case 2: {
+        _power = 0.8;
+        _compensation = 8.5;
+        _frequency = 50;
+        _eqSound = "Earthquake_02";
+    };
+    case 3: {
+        _power = 1.5;
+        _compensation = 7;
+        _frequency = 40;
+        _eqSound = "Earthquake_03";
+    };
+    case 4: {
+        _power = 2.1;
+        _compensation = 6;
+        _frequency = 30;
+        _eqSound = "Earthquake_04";
+    };
+    case 5: {
+        _power = 2.5;
+        _compensation = 8;
+        _frequency = 25;
+        _eqSound = "Earthquake_04";
+    };
+    default {
+        if (is3DENPreview) then {
+            hint format ["[Earthquake]: Magnitude (%1) not between 1-5, using default values.", _magnitude];
```

Comment:
This should be printed outside 3DEN as well, as a warning. The hint above as well.

Reply:
I'll redo all errors for next release, find a nice method.